src/main/java/io/cdap/plugin/gcp/bigquery/sink/AbstractBigQuerySinkConfig.java

```diff
@@ -138,5 +144,12 @@ public void validate(FailureCollector collector) {
     if (!containsMacro(NAME_DATASET)) {
       BigQueryUtil.validateDataset(dataset, NAME_DATASET, collector);
     }
+
+    //note - These options could be set together for write to single partition in the future
+    if (isTruncateTableSet() && isAllowSchemaRelaxation()) {
```

Comment:
I don't understand why these can't be used together. From the comment earlier, I assume that when the table is being truncated, the schema updates automatically.

Reply:
If both are set, the BQ load fails with this message  - "Could not commit Job: Error occurred while importing data to BigQuery 'Schema update options should only be specified with WRITE_APPEND disposition, or with WRITE_TRUNCATE disposition on a table partition.' 
" .  https://b.corp.google.com/issues/155237532#comment7 gives some more insights as to why they are incompatible.

Reply:
that means we should not be setting both of those options for the BQ load, but it doesn't mean the user shouldn't be able to set these at the plugin level. From the user's perspective, the schema can be updated when the table is truncated, as it can be updated to anything. 

The fix should be that we don't set schema update options for the load job if the write disposition doesn't allow it.

Reply:
Changed to ensure schema update options are not set if write disposition is WRITE_TRUNCATE. User will not be given a validation error